src/ngraph/op/reverse.cpp

```diff
@@ -69,3 +70,132 @@ void op::Reverse::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVect
 
     adjoints.add_delta(x, make_shared<op::Reverse>(delta, m_reversed_axes));
 }
+
+const string op::v1::Reverse::type_name{"Reverse"};
+
+op::v1::Reverse::Reverse(const Output<Node>& data,
+                         const Output<Node>& reversed_axes,
+                         const std::string& mode)
+    : Op({data, reversed_axes})
+    , m_mode{mode_from_string(mode)}
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v1::Reverse::Reverse(const Output<Node>& data,
+                         const Output<Node>& reversed_axes,
+                         const Mode mode)
+    : Op({data, reversed_axes})
+    , m_mode{mode}
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v1::Reverse::validate_and_infer_types()
+{
+    if (m_mode == Mode::MASK)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              get_input_element_type(1) == element::boolean,
+                              "In 'mask' mode the second input must contain boolean values.");
+    }
+
+    const auto input_shape = get_input_partial_shape(0);
+    const auto input_rank = input_shape.rank();
+
+    const auto rev_axes_shape = get_input_partial_shape(1);
+    const auto rev_axes_rank = rev_axes_shape.rank();
+
+    if (rev_axes_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<size_t>(rev_axes_rank) == 1,
+                              "The reversed_axes input must be a 1D tensor (got ",
+                              static_cast<size_t>(rev_axes_rank),
+                              ").");
+
+        if (m_mode == Mode::MASK)
+        {
+            NODE_VALIDATION_CHECK(
+                this,
+                static_cast<size_t>(rev_axes_rank) == 1,
+                "The reversed_axes input rank needs to be equal to 1 in 'mask' mode.");
```

Comment:
In 'index' mode it should be 1 as well. Why not to check outside m_mode == Mode::MASK condition?